text/3579-pki.md

```diff
@@ -0,0 +1,262 @@
+
+
+### Staying Provider Agnostic
+
+This RFC attempts to limit our exposure to singular points of failure or compromise by relying on entities for the totality of our security. A choice was made to utilize Amazon CloudHSM; however, we are capable of moving the entirety of this operation to another infrastructure if needed. Some cases here can be solved by various cloud providers, which our own CA would allow us to remain independent; such as internal mTLS, binary code signing and git commit signatures.
+
+
+# Prior art
+[prior-art]: #prior-art
+
+[IANA uses a quorum system and "signing ceremony" for the root keys of DNSSEC](https://www.iana.org/dnssec/).
+
+[Sigstore performs a signing ceremony among multiple root signers](https://github.com/sigstore/root-signing).
+
+`kernel.org` was compromised, which forced them to evaluate and address their threat model. They've assessed their threat model to include very well-funded security threats, as well as nation-state-level legal attacks and security threats. In this RFC, we're proposing an infrastructure that addresses comparable threat models.
+https://lwn.net/Articles/464233/
+https://www.theregister.com/2011/08/31/linux_kernel_security_breach/
+https://arstechnica.com/information-technology/2013/09/who-rooted-kernel-org-servers-two-years-ago-how-did-it-happen-and-why/
+
+
+Relevant past RFCs:
+
+[RFC #3403: sigstore and cargo/crates.io](https://github.com/rust-lang/rfcs/pull/3403)
+[TUF for crates.io discussion](https://github.com/withoutboats/rfcs/pull/7)
+[Validate signatures of builds](https://github.com/rust-lang/rustup/issues/241)
+
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Who shall be members of the Quorum? This RFC leaves this question to the Leadership Council. The quorum could consist of Council members, or the Council can choose to delegate authority to trusted indivuals, such as the existing Security Response or Secure Code working groups; or the formation of a new Security/PKI working group with a limited scope of managing these keys and performing ceremonies.
+
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+**Crate Signing / Mirroring**: A subsequent RFC will specify how we handle crates.io index signing, to enable mirroring and other requirements.
+
+**Code Signing**: This RFC provides a chain of trust that can in the future be used for signing binaries for Apple/Microsoft binary authentication. However, this RFC does not specify a mechanism or process for such signing; establishing that is future work. A subsequent RFC will specify how we handle code signing.
```

Comment:
How is code signing related to this RFC? Windows code signing uses commercially purchased certificates, and Apple code signing uses Apple-issued certificates.

Reply:
Addressed in the changes we just pushed. It's possible to generate a CSR from our own private key that chains up to the Rust root, and get Apple/Microsoft to sign that CSR for use with code signing.